kge/job/eval.py

```diff
@@ -141,3 +146,66 @@ def create_from(
             new_config.set("eval.split", eval_split, create=True)
 
         return super().create_from(checkpoint, new_config, dataset, parent_job)
+
+
+class TrainingLossEvaluationJob(EvaluationJob):
+    """ Evaluating simply by using the training loss """
+
+    def __init__(self, config: Config, dataset: Dataset, parent_job, model):
+        super().__init__(config, dataset, parent_job, model)
+
+        training_loss_eval_config = config.clone()
+        training_loss_eval_config.set("console.quiet", False)
+        # TODO set train split to include validation data here
+        #   once support is added
+        #   Then reflect this change in the trace entries
+
+        self._train_job = TrainingJob.create(
+            config=training_loss_eval_config,
+            parent_job=self,
+            dataset=dataset,
+            model=model,
+            forward_only=True,
+        )
+
+        # prepare training job
+        self._train_job._prepare()
+        self._train_job._is_prepared = True
+
+        if self.__class__ == TrainingLossEvaluationJob:
+            for f in Job.job_created_hooks:
+                f(self)
+
+    def _prepare(self):
+        super()._prepare()
+
+    @torch.no_grad()
+    def _evaluate(self) -> Dict[str, Any]:
+        if self.parent_job:
+            self.epoch = self.parent_job.epoch
+
+        # create initial trace entry
+        self.current_trace["epoch"] = dict(
+            type="training_loss",
+            scope="epoch",
+            # split=self.eval_split,
+            split=self._train_job.config.get("train.split"),
+            epoch=self.epoch,
+        )
+
+        # run pre-epoch hooks (may modify trace)
+        for f in self.pre_epoch_hooks:
+            f(self)
+
+        # let's go
+        train_trace_entry = self._train_job.run_epoch()
```

Comment:
metric computation is missing here. in particular, if the metric is not in the trace, the metric expr needs to be computed. the code that does this should be moved to our metric class

Reply:
My bad, this is invalid!

Reply:
But it does raise an interesting issue: what should be added to current_trace by this type of eval job? I guess simply the training job trace? Currently, it is "harcoded" to return avg loss and sum loss.

Reply:
I guess we should add the event eval completed and done. Then custom metrics can be defined on the train trace

Reply:
good question. for now, we could trace the train entry separately (as we do for valid) and take over the loss, penalty, cost to the eval job.